src/forEach.js

```diff
@@ -30,4 +31,6 @@ var _forEach = require('./internal/_forEach');
  *      //-> 7
  *      //-> 8
  */
-module.exports = _curry2(_forEach);
+module.exports = _curry2(function forEach(fn, list) {
+  return _hasMethod('forEach', list) ? list.forEach(fn) : _forEach(fn, list);
```

Comment:
> But if we dispatch `forEach` to `Map`/`Set`, `undefined` will be returned.

That's a problem with all dispatching.  We can't guarantee that the return of the dispatched-to function matches our own.  The user is really responsible for that.

In this case, it's really not a big deal, to my eyes, as `forEach` has no logical return.  Ramda simply returns the original list in order to return _something_, not because there's any great virtue in that particular return.